Orthofinder ouptus a lot of useful stats. Almost everything you need can be found within the directory Comparative_Genomics_Statistics.

These 3 files were used to get most stats:
* Orthogroups_SpeciesOverlaps.tsv
* Statistics_PerSpecies.tsv
* OrthologuesStats_Totals.tsv


### Scripts

* countGeneSpace.sh - count bp that genes occupy in gtf.
* getPrimaryTranscripts.sh - Get all primary transcripts from a faa file (amino acid fasta).
* makeGenesFile.sh - Create csv with gene locations and name from breaker2 gtf.
* orthogrouping.sh - Perform ortho grouping with OrthoFinder
* panGeneCounts.sh - Count the number of core, dispensible, and private genes per genome.
